src/microcosm.js

```diff
@@ -206,24 +206,44 @@ class Microcosm extends Emitter implements Domain {
    * that, when rolling back to this action, it always has a state value.
    */
   createSnapshot(action: Action) {
-    this.snapshots[action.id] = this.recall(action.parent)
+    let state = this.recall(action.parent)
+
+    let snapshot: Snapshot = {
+      last: state,
+      next: state,
+      status: action.status,
+      payload: action.payload
+    }
+
+    this.snapshots[action.id] = snapshot
+
+    return snapshot
   }
 
   /**
    * Update the state snapshot for a given action
    */
   updateSnapshot(action: Action) {
-    let next = this.recall(action.parent)
+    // Fall back to creating a snapshot if it does not exist in the event
+    // an action is in progress while a fork is being created
+    let snap = this.snapshots[action.id] || this.createSnapshot(action)
+    let last = this.recall(action.parent)
```

Comment:
Ahh. Yes. We can do this. It looks like the `START` action that kicks things off is never getting assigned to a snapshot state.

That's a bummer. Adding START allowed me to avoid some conditional logic, but now it's causing problems. Maybe I can figure out how to remove it.